hmftools.wdl

```diff
@@ -0,0 +1,927 @@
+        amberOutput: {description: "The output files of hmftools amber.", category: "required"}
+        cobaltOutput: {description: "The output files of hmftools cobalt", category: "required"}
+        gcProfile: {description: "A file describing the GC profile of the reference genome.", category: "required"}
+        somaticVcf: {description: "The somatic variant calling results.", category: "required"}
+        germlineVcf: {description: "The germline variant calling results.", category: "required"}
+        filteredSvVcf: {description: "The filtered structural variant calling results.", category: "required"}
+        fullSvVcf: {description: "The unfiltered structural variant calling results.", category: "required"}
+        referenceFasta: {description: "The reference fasta file.", category: "required"}
+        referenceFastaDict: {description: "The sequence dictionary associated with the reference fasta file.",
+                             category: "required"}
+        referenceFastaFai: {description: "The index for the reference fasta file.", category: "required"}
+        driverGenePanel: {description: "A TSV file describing the driver gene panel.", category: "required"}
+        somaticHotspots: {description: "A vcf file with hotspot somatic variant sites.", category: "required"}
+        germlineHotspots: {description: "A vcf file with hotspot germline variant sites.", category: "required"}
+
+        threads: {description: "The number of threads the program will use.", category: "advanced"}
+        memory: {description: "The amount of memory this job will use.", category: "advanced"}
+        javaXmx: {description: "The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.",
+                  category: "advanced"}
+        timeMinutes: {description: "The maximum amount of time the job will run in minutes.", category: "advanced"}
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.",
+                      category: "advanced"}
+    }
+}
+
+task Sage {
+    input {
+        String tumorName
+        File tumorBam
+        File tumorBamIndex
+        File referenceFasta
+        File referenceFastaDict
+        File referenceFastaFai
+        File hotspots
+        File panelBed
+        File highConfidenceBed
+        Boolean hg38 = false
+        Boolean panelOnly = false
+        String outputPath = "./sage.vcf.gz"
+
+        String? normalName
+        File? normalBam
+        File? normalBamIndex
+        Int? hotspotMinTumorQual
+        Int? panelMinTumorQual
+        Int? hotspotMaxGermlineVaf
+        Int? hotspotMaxGermlineRelRawBaseQual
+        Int? panelMaxGermlineVaf
+        Int? panelMaxGermlineRelRawBaseQual
+        String? mnvFilterEnabled
+        File? coverageBed
+
+        Int threads = 4
+        String javaXmx = "50G"
+        String memory = "60G"
```

Comment:
If javaxmx = 50G the memory should not exceed 51G. There is no piping and there are no other commands active.